addons/mail/models/mail_tracking_value.py

```diff
@@ -88,34 +88,49 @@ def create_tracking_values(self, initial_value, new_value, col_name, col_info, t
             return values
         return {}
 
-    def get_display_value(self, type):
-        assert type in ('new', 'old')
+    def _tracking_value_format(self):
+        tracking_values = [{
+            'changedField': tracking.field_desc,
+            'id': tracking.id,
+            'newValue': [('insert-and-replace', {
+                'currencyId': tracking.currency_id.id,
+                'fieldType': tracking.field_type,
+                'value': tracking._get_new_display_value()[0]})],
+            'oldValue': [('insert-and-replace', {
+                'currencyId': tracking.currency_id.id,
+                'fieldType': tracking.field_type,
+                'value': tracking._get_old_display_value()[0]})],
+        } for tracking in self]
+        return [('insert-and-replace', tracking_values)] if tracking_values else []
```

Comment:
Actually the `insert-and-replace` should probably be in the parent instead. This should just return an array of tracking data, the fact that it is inserted into a field is message concern.